src/AutoRest.CSharp.V3/Generation/Writers/JsonCodeWriterExtensions.cs

```diff
@@ -308,45 +309,73 @@ private static void CollectProperties(Dictionary<ObjectTypeProperty, CodeWriterD
             }
         }
 
-        private static void DeserializeIntoVariable(this CodeWriter writer, JsonSerialization serialization, Action<CodeWriter, CodeWriterDelegate> valueCallback, CodeWriterDelegate element)
+        private static void DeserializeIntoVariable(this CodeWriter writer, JsonSerialization serialization, Action<CodeWriter, CodeWriterDelegate> valueCallback,
+            CodeWriterDelegate element, bool? writeNullHandling = null)
         {
+            void WriteNullableScope(Action writeContent)
+            {
+                if (writeNullHandling == true)
+                {
+                    using (writer.Scope($"if ({element}.ValueKind == {typeof(JsonValueKind)}.Null)"))
+                    {
+                        valueCallback(writer, w => w.Append($"null"));
+                    }
+                    using (writer.Scope($"else"))
+                    {
+                        writeContent();
+                    }
+                }
+                else
+                {
+                    writeContent();
+                }
+            }
+
             switch (serialization)
             {
                 case JsonArraySerialization array:
-                    var arrayVariable = new CodeWriterDeclaration("array");
-                    writer.Line($"{array.Type} {arrayVariable:D} = new {array.Type}();");
-
-                    var collectionItemVariable = new CodeWriterDeclaration("item");
-                    writer.Line($"foreach (var {collectionItemVariable:D} in {element}.EnumerateArray())");
-                    using (writer.Scope())
+                    writeNullHandling ??= true;
```

Comment:
Is it correct to inherit this value for inner serializations?

Reply:
By 'inherit', you mean 'use the value provided by the parameter', then yes. This is how it works for property serializers, where they provide `false` to this method, and it acknowledges it appropriately. Otherwise, if a value is not provided (null), these act like non-value types, thus `true` to writing the null handler.

Reply:
I don't believe so as inner serializations will have their own calls to this method that would have their own value passed in for this. I originally thought I had to do this, but our `property bag` recursive logic seems to handle this correctly. Meaning, inner properties will be written flat in the main property loop, so there isn't a 'depth' to keeping track of this value.

Reply:
What about the List<List<int>> case?

Reply:
So, the inner `List<int>` isn't a property of the outer type, so the `continue;` null check isn't written to begin with. Since it is a non-value type, it should write the new null check. Similar example here in `List<Fish>` from Sawshark.Serializartion.cs in body-complex.

![image](https://user-images.githubusercontent.com/17788297/78390567-a2ccd380-7599-11ea-9293-8f7e4957446b.png)